nes/api/app.py

```
@@ -54,10 +54,32 @@ async def lifespan(app: FastAPI):
+    # Initialize the search backend (if configured) and ensure its index exists.
+    # If the backend is unreachable, degrade gracefully: detach it from the
+    # search service so requests fall back to database search instead of failing.
+    backend = config.Config.get_search_backend()
+    if backend is not None:
+        try:
+            if await backend.health():
+                await backend.ensure_index()
+                logger.info("Search backend ready: %s", type(backend).__name__)
+            else:
+                raise RuntimeError("search backend health check failed")
+        except Exception as e:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)